lib/administrate/search.rb

```diff
@@ -31,6 +38,24 @@ def search_attributes
       end.keys
     end
 
-    attr_reader :resolver, :term
+    def search_scope(term)
+      if (term[-1, 1] == ':')
+        possible_scope = term[0..-2]
+        possible_scope if resource_class.respond_to?(possible_scope) and
+                          not banged?(possible_scope) and
+                          not blacklisted_scope?(possible_scope)
+      end
+    end
+
+    def banged?(method)
+      method[-1, 1] == '!'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.